test/python/circuit/test_initializer.py

```diff
@@ -367,6 +367,44 @@ def test_from_int(self):
         actual_sv = Statevector.from_instruction(qc)
         self.assertTrue(desired_sv == actual_sv)
 
+    def _remove_resets(self, circ):
+        circ.data = [tup for tup in circ.data if tup[0].name != 'reset']
+
+    def test_global_phase_random(self):
+        """Test global phase preservation with random state vectors"""
+        from qiskit.quantum_info.random import random_statevector
+        repeats = 5
+        for n_qubits in [1, 2, 4]:
+            for irep in range(repeats):
+                with self.subTest(i=f'{n_qubits}_{irep}'):
+                    dim = 2**n_qubits
+                    qr = QuantumRegister(n_qubits)
+                    initializer = QuantumCircuit(qr)
+                    target = random_statevector(dim)
+                    initializer.initialize(target, qr)
+                    uninit = initializer.data[0][0].definition
+                    self._remove_resets(uninit)
+                    disentangler = Operator(uninit.data[0][0])
+                    zero = Statevector.from_int(0, dim)
+                    evolve = zero @ disentangler
```

Comment:
I think you can just do
```suggestion
                    evolve = Statevector(uninit)
```
since the `Statevector` initializer supports circuits 🙂